src/App.jsx

```diff
@@ -1,14 +1,38 @@
+import { useState } from 'react';
 import './App.css';
+import ChatLog from './components/ChatLog';
+import DATA from './data/messages.json';
 
+// const
 const App = () => {
+  const [chatData, setChatData] = useState(DATA);
+
+  const handleLikeClick = (id) => {
+    setChatData((prevChatData) => {
+      const updatedChat = prevChatData.map((chat) => {
+        if (chat.id === id) {
+          const isNowLiked = !chat.liked; // Toggle the like status
+          return { ...chat, liked: isNowLiked };
+        }
+        return chat;
+      });
+      return updatedChat;
+    });
+  };
+  const likedCount = chatData.filter((chat) => chat.liked).length;
```

Comment:
Consider adding a blank line between lines 21 and 22 to emphasize that these are two logically different pieces of code.

```suggestion
  };
  
  const likedCount = chatData.filter((chat) => chat.liked).length;
```

Reply:
I'd prefer this logic to be in a helper function even though it's not complex. Then you can invoke the function on line 29, maybe something like `getLikeCount`.

Also, you can use `reduce` to find the number of likes:
```js
const getLikeCount = () => {
  // adding a bool to a number treats true as 1 and 0 as false
  return chatData.reduce((totalLikes, chat) => (totalLikes + chat.liked), 0);
};

```